backend/src/search/search.controller.ts

```diff
@@ -0,0 +1,38 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import { SearchService } from './search.service';
+import { RequiredPipe } from '../pipes/RequiredPipe';
+import { SearchResult } from '../shared/search/SearchResult';
+
+@Controller('search')
+export class SearchController {
+  constructor(private readonly searchService: SearchService) {}
+
+  /**
+   * Filters through all properties of all nodes and all edges (case-insensitive) .
+   * @param filter filter string ("if node/edge-property contains filter -> return it")
+   *
+   * @example
+   * 'Keanu' returns 'Keanu Reeves'
+   * @example
+   * 'kean' returns 'Keanu Reeves'
+   * @example
+   * 'reev' returns 'Keanu Reeves'
+   * @example
+   * 'name' returns 'Keanu Reeves' (since this node has the attribute 'name')
+   */
+  @Get('all')
+  all(@Query('filter', RequiredPipe) filter: string): Promise<SearchResult> {
+    return this.searchService.search(filter);
```

Comment:
instead of piping `filter` through `RequiredPipe` would it make sense to use nestjs's ´@IsNotEmpty()` class-validator? See [auto-validation example](https://docs.nestjs.com/techniques/validation#auto-validation) at https://docs.nestjs.com/techniques/validation

Reply:
It might be possible, but I don't know if it works on primitives as well, the example is only for classes/class-attributes.
If it's okay for you I'd merge it.

Reply:
Yes, it's okay for me